schematools/cli.py

```diff
@@ -144,68 +148,53 @@ def introspect_db(prefix, db_url, dataset_id, tables):
 
 
 @import_.command("ndjson")
-@click.option(
-    "--db-url",
-    envvar="DATABASE_URL",
-    required=True,
-    help="DSN of database, can also use DATABASE_URL environment.",
-)
-@click.option(
-    "--schema-url",
-    envvar="SCHEMA_URL",
-    default=DEFAULT_SCHEMA_URL,
-    show_default=True,
-    required=True,
-    help="Url where valid amsterdam schema files are found. "
-    "SCHEMA_URL can also be provided as environment variable.",
-)
-@click.argument("dataset_id")
+@option_db_url
+@option_schema_url
+@argument_schema_location
 @click.argument("table_name")
 @click.argument("ndjson_path")
-def import_ndjson(db_url, schema_url, dataset_id, table_name, ndjson_path):
+def import_ndjson(db_url, schema_url, schema_location, table_name, ndjson_path):
     """Import an NDJSON file into a table."""
     # Add batching for rows.
     engine = _get_engine(db_url)
-    dataset_schema = schema_def_from_url(schema_url, dataset_id)
+    dataset_schema = schema_def_from_url(schema_url, schema_location)
     srid = dataset_schema["crs"].split(":")[-1]
     with open(ndjson_path) as fh:
         data = list(fetch_rows(fh, srid))
     create_rows(engine, metadata, dataset_schema, table_name, data)
 
 
 @import_.command("schema")
-@click.option(
-    "--db-url",
-    envvar="DATABASE_URL",
-    required=True,
-    help="DSN of database to write into, can also use DATABASE_URL environment.",
-)
-@click.option(
-    "--schema-url",
-    envvar="SCHEMA_URL",
-    default=DEFAULT_SCHEMA_URL,
-    show_default=True,
-    required=True,
-    help="Url where valid amsterdam schema files are found. "
-    "SCHEMA_URL can also be provided as environment variable.",
-)
-@click.argument("dataset_id")
-def import_schema(db_url, schema_url, dataset_id):
+@option_db_url
+@option_schema_url
+@argument_schema_location
+def import_schema(db_url, schema_url, schema_location):
     """Import the schema definition into the local database."""
     # Add drop or not flag
     engine = _get_engine(db_url)
-    try:
-        dataset_schema = schema_def_from_url(schema_url, dataset_id)
-    except KeyError:
-        raise click.BadParameter(f"Schema {dataset_id} not found.")
+    dataset_schema = _get_dataset_schema(schema_url, schema_location)
 
     create_meta_tables(engine)
     create_meta_table_data(engine, dataset_schema)
+
+
+def _get_dataset_schema(schema_url, schema_location) -> DatasetSchema:
+    """Find the dataset schema for the given dataset"""
+    if "." in schema_location or "/" in schema_location:
+        click.echo(f"Reading schema from {schema_location}")
+        return DatasetSchema.from_file(schema_location)
+    else:
+        # Read the schema from the online repository.
+        click.echo(f"Reading schemas from {schema_url}")
+        try:
+            return schema_def_from_url(schema_url, schema_location)
+        except KeyError:
+            raise click.BadParameter(f"Schema {schema_location} not found.")
```

Comment:
Misschien de echo() calls niet default, maar alleen bij --verbose (die we dan nog toe moeten voegen :-).  Het is een beetje tegen de unix tool filosofie om output te genereren die niet met de data zelf te maken heeft als de gebruiker er niet om vraagt:
https://en.wikipedia.org/wiki/Unix_philosophy#Origin (punt 2)

Alhoewel talloze cli tools hiertegen zondigen, misschien ben ik ouderwets ;-)

Reply:
@jjmurre deze snap ik niet helemaal. Bedoel je dat dit geen `schema_location` moet tonen? De exception komt op stderr terecht, met een exit 1.